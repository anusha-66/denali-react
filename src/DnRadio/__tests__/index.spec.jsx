/**
 * Copyright 2020, Verizon Media
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */

import React from 'react';
import { render, mount } from 'enzyme';
import { DnRadio } from '../../index'; // src/index.tsx

describe('DnCheckbox Tests', () => {
    it('should render with default props', () => {
        expect.assertions(2);

        const wrapper = mount(<DnRadio id="Radio" />);

        expect(wrapper.render()).toMatchSnapshot('before change');

        wrapper.find('input[type="radio"]').simulate('change');

        expect(wrapper.render()).toMatchSnapshot('after change');
    });

    it('should render correctly with custom props', () => {
        expect.assertions(1);

        const rendered = render(
            <>
                <DnRadio
                    id="Radio1"
                    name="sample"
                    radioValue="Radio1"
                    defaultChecked
                    isDisabled
                    className="additional--class"
                    radioLabel="Radio 1"
                />
                <DnRadio id="Radio2" name="sample" radioValue="Radio2" defaultChecked isDisabled radioLabel="Radio 2" />
                <DnRadio id="Radio3" name="sample" radioValue="Radio3" defaultChecked isDisabled radioLabel="Radio 3" />
            </>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should call on change correctly', () => {
        expect.assertions(1);

        const mockedOnChange = jest.fn();
        const wrapper = mount(<DnRadio onChange={mockedOnChange} />);
        wrapper.find('input[type="radio"]').simulate('change');

        expect(mockedOnChange).toHaveBeenCalledTimes(1);
    });
});
