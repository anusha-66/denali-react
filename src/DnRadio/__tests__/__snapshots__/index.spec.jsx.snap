// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DnCheckbox Tests should render correctly with custom props 1`] = `
Array [
  <div
    class="radio additional--class"
  >
    <input
      checked=""
      disabled=""
      id="Radio1"
      name="sample"
      type="radio"
      value="Radio1"
    />
    <label
      for="Radio1"
    >
      Radio 1
    </label>
  </div>,
  <div
    class="radio"
  >
    <input
      checked=""
      disabled=""
      id="Radio2"
      name="sample"
      type="radio"
      value="Radio2"
    />
    <label
      for="Radio2"
    >
      Radio 2
    </label>
  </div>,
  <div
    class="radio"
  >
    <input
      checked=""
      disabled=""
      id="Radio3"
      name="sample"
      type="radio"
      value="Radio3"
    />
    <label
      for="Radio3"
    >
      Radio 3
    </label>
  </div>,
]
`;

exports[`DnCheckbox Tests should render with default props: after change 1`] = `
<div
  class="radio"
>
  <input
    id="Radio"
    type="radio"
    value=""
  />
  <label
    for="Radio"
  />
</div>
`;

exports[`DnCheckbox Tests should render with default props: before change 1`] = `
<div
  class="radio"
>
  <input
    id="Radio"
    type="radio"
    value=""
  />
  <label
    for="Radio"
  />
</div>
`;
